### PR TITLE
RUMM-186 Bundle the Log with the current active Trace

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

This PR bundles(links) the logs with the the active Trace in the application. Having this relation between the log and the current active span(trace) we will be able to see the logs sent for a specific trace. For now the **spanId** parameter is not used but probably will be used later so we added it anyway.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/test/kotlin/com/datadog/android/log/LoggerBuilderTest.kt [hunk 1]
```diff
@@ -164,4 +165,16 @@ internal class LoggerBuilderTest {
         val handler: DatadogLogHandler = logger.getFieldValue("handler")
         assertThat(handler.loggerName).isEqualTo(loggerName)
     }
+
+    @Test
+    fun `buider can disable the bundle with trace feature`(@Forgery forge: Forge) {
```
Comment (ncreated) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/log/LoggerBuilderTest.kt]: Typo: `builder`

File: dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/utils/TracerExtensionsTest.kt [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        val tracer = GlobalTracer.get()
+        val span = tracer.buildSpan(forge.anAlphabeticalString()).start() as DDSpan
+        tracer.activateSpan(span)
+
+        // then
+        assertThat(tracer.traceId()).isEqualTo(span.traceId.toString())
+        assertThat(tracer.spanId()).isEqualTo(span.spanId.toString())
+    }
+    @Test
+    fun `it will return null for trace an span id if there is no active span`(forge: Forge) {
```
Comment (ncreated) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/tracing/internal/utils/TracerExtensionsTest.kt]: Typo: `and`

File: dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandlerTest.kt [hunk 3]
```diff
@@ -214,4 +229,86 @@ internal class DatadogLogHandlerTest {
...
     }
+
+    @Test
+    fun `it will add the span id and trace id if we active an active tracer`(forge: Forge) {
+        // given
+        val config = DatadogConfig.Builder(forge.anAlphabeticalString()).build()
+        Datadog.initialize(mockContext(), config)
+        val tracer = Tracer.Builder().build()
+        val span = tracer.buildSpan(forge.anAlphabeticalString()).start()
+        tracer.activateSpan(span)
```
Comment (ncreated) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandlerTest.kt]: What will happen if we don't activate Span? I think we'll end up with `dd.trace_id` set but `dd.span_id` unset for the serialized log. Is this valid scenario?

Comment (mariusc83) [reply dd-sdk-android/src/test/kotlin/com/datadog/android/log/internal/logger/DatadogLogHandlerTest.kt]: If you check the way the trace_id and span_id are generated in the TracerExtensions ...there's no way to have one without the other. If there's no active span the trace_id and span_id will be null. Everything is taken from the activeSpan. I followed exactly what they were doing in the Java API.

**Pull request merged**